pkg/database/service.go

```diff
@@ -266,7 +270,66 @@ func (s service) FindExternalDownload(uuid uuid.UUID) (model.ExternalDownload, e
 	return s.repository.FindExternalDownload(uuid)
 }
 
-func (s service) SaveAs(database *model.Database, instance *model.Instance, stack *model.Stack, newName string, format string) (*model.Database, error) {
+func (s service) Save(database *model.Database, instance *model.Instance, stack *model.Stack) error {
+	tmpName := randomString(20)
+	format := getFormat(database)
+	_, err := s.SaveAs(database, instance, stack, tmpName, format, func(saved *model.Database) {
+		err := s.Delete(database.ID)
+		if err != nil {
+			logError(err)
+			return
+		}
+
+		u, err := url.Parse(saved.Url)
```

Comment:
We could move this before `s.Delete` so we don't delete in case we cannot parse the new saved DB 🤔